addon/-private/formatters/format-message.ts

```diff
@@ -52,19 +52,29 @@ function escapeOptions<T extends Record<string, unknown>>(object?: T) {
 /**
  * @private
  * @hide
- * @TODO Should not extend from abstract base class.
  */
-export default class FormatMessage extends Formatter<Options> {
-  // @ts-ignore This class does not match the abstract base class.
+export default class FormatMessage {
+  static readonly type = 'message';
+
+  protected readonly config: FormatterConfig;
+  protected readonly readFormatConfig: () => Formats;
+
+  constructor(config: FormatterConfig) {
+    this.config = config;
+
+    // NOTE: a fn since we lazily grab the formatter from the config
+    // as it can change at runtime by calling intl.set('formats', {...});
+    this.readFormatConfig = config.readFormatConfig;
+  }
+
   createNativeFormatter = memoize(
     (ast: TranslationAST, locales: string | string[], formatConfig?: Partial<Formats>) => {
-      return new IntlMessageFormat(ast as any, locales, formatConfig, {
+      return new IntlMessageFormat(ast, locales, formatConfig, {
```

Comment:
https://github.com/ember-intl/ember-intl/pull/1420#discussion_r471896783

Weirdly, the issue is not occuring any more. Maybe there was a dependency version mismatch?

Reply:
I upgraded the intl-related deps in 1214587e493fdf08f4570e7b7cc3cc11ff2f1ad0 just be _really_ sure 😅